In [1]:
import pandas as pd
import numpy as np
from IPython.display import Image
import matplotlib.pyplot as plt
import seaborn as sns

import os
for dirname, _, filenames in os.walk('Data'):
    for filename in filenames:
        print(os.path.join(dirname, filename))

Data\Olympics_Country.csv
Data\Olympics_Games.csv
Data\Olympic_Athlete_Bio.csv
Data\Olympic_Athlete_Event_Results.csv
Data\Olympic_Results.csv
Data\Pipfile


### Olympic Results

The result table is used to keep track of who won, what sport and in which location

##### Primary Key => resultID

In [2]:
path_Olympic_OlympicResults ="Data\Olympic_Results.csv"

In [3]:
OR = pd.read_csv(path_Olympic_OlympicResults, sep =",")
OR.head()

result_id                         event_title               edition  \
0       9270         7 metres, 1907 Rating, Open  1920 Summer Olympics   
1      29722  Featherweight (≤60 kilograms), Men  1980 Summer Olympics   
2      54126                       Platform, Men  1948 Summer Olympics   
3       4684          100 metres Backstroke, Men  1920 Summer Olympics   
4       1465                         Slalom, Men  2002 Winter Olympics   

   edition_id          sport                sport_url  \
0           7        Sailing   /editions/7/sports/SAL   
1          20  Weightlifting  /editions/20/sports/WLF   
2          12         Diving  /editions/12/sports/DIV   
3           7       Swimming   /editions/7/sports/SWM   
4          47  Alpine Skiing  /editions/47/sports/ALP   

                                     result_location   result_participants  \
0                                           Oostende    8 from 2 countries   
1                   Dvorets sporta Izmaylovo, Moskva  18 from 14 countries   
2                     Wembley Arena, Wembley, London  25 from 15 countries   
3               Zwemstadion van Antwerpen, Antwerpen   12 from 6 countries   
4  Park City Mountain Resort, Park City, Utah (Kn...  77 from 43 countries   

                                       result_format result_detail  \
0  Three races, with final places decided by tota...            na   
1  Total of best lifts in snatch and clean & jerk...            na   
2  10 metre platform. Final round only. Four comp...            na   
3                                                 na            na   
4         Two runs, total time determined placement.    Jesse Hunt   

                                  result_description  start_date    end_date  \
0  There were two yachts competing.  Britain’s An...  1920-07-07  1920-07-09   
1  The favorite was likely Soviet Viktor Mazin wh...  1980-07-22          na   
2  Sammy Lee won this event, leading in both the ...  1948-08-04  1948-08-05   
3  Watten Paoa Kealoha set a world record of 1:14...  1920-08-22  1920-08-23   
4  The last few World Cups had gone to Austrian B...  2002-02-23          na   

          time  
0           na  
1  14:00-19:00  
2           na  
3           na  
4           na

In [22]:
OR.edition.unique().shape
# The olympics datasetis for summer olympics  from 1920 and 2022
# The Summer Olympic Games and Winter Olympic Games are each held every four years

(55,)

In [17]:
OR.info()
# Does not have any null

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7329 entries, 0 to 7328
Data columns (total 14 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   result_id            7329 non-null   int64 
 1   event_title          7329 non-null   object
 2   edition              7329 non-null   object
 3   edition_id           7329 non-null   int64 
 4   sport                7329 non-null   object
 5   sport_url            7329 non-null   object
 6   result_location      7329 non-null   object
 7   result_participants  7329 non-null   object
 8   result_format        7329 non-null   object
 9   result_detail        7329 non-null   object
 10  result_description   7329 non-null   object
 11  start_date           7329 non-null   object
 12  end_date             7329 non-null   object
 13  time                 7329 non-null   object
dtypes: int64(2), object(12)
memory usage: 801.7+ KB


In [12]:
# Size of the dataset
OR.shape

(7329, 14)

#### Necessary Columns

In the above dataset, I would suggest dropping any columns that are not necessary in building a business case. These are:
    sport_url, result_participants, result_format, result_description, 

In [4]:
OR["event_title"].unique().shape

(1555,)

Seeing that there are more that 1555 event titles, I would consider dropping this column too as a I dont see any 

In [5]:
ORmod = OR.drop(columns = ['sport_url', 'result_participants', 'result_format', 'result_description', 'event_title'])
ORmod.head()
## OR modified

result_id               edition  edition_id          sport  \
0       9270  1920 Summer Olympics           7        Sailing   
1      29722  1980 Summer Olympics          20  Weightlifting   
2      54126  1948 Summer Olympics          12         Diving   
3       4684  1920 Summer Olympics           7       Swimming   
4       1465  2002 Winter Olympics          47  Alpine Skiing   

                                     result_location result_detail  \
0                                           Oostende            na   
1                   Dvorets sporta Izmaylovo, Moskva            na   
2                     Wembley Arena, Wembley, London            na   
3               Zwemstadion van Antwerpen, Antwerpen            na   
4  Park City Mountain Resort, Park City, Utah (Kn...    Jesse Hunt   

   start_date    end_date         time  
0  1920-07-07  1920-07-09           na  
1  1980-07-22          na  14:00-19:00  
2  1948-08-04  1948-08-05           na  
3  1920-08-22  1920-08-23           na  
4  2002-02-23          na           na

Here above shoud be IMO the final dataset for the olympic results.

##### Unavailable data
StartDate, EndDate and Time

In [13]:
#naOR = ORmod.loc[np.logical_and(ORmod.start_date == "na", ORmod.end_date == "na", ORmod.time == "na")]
naOR = ORmod.loc[ORmod.result_detail == "na"]
naOR.head()

result_id               edition  edition_id          sport  \
0       9270  1920 Summer Olympics           7        Sailing   
1      29722  1980 Summer Olympics          20  Weightlifting   
2      54126  1948 Summer Olympics          12         Diving   
3       4684  1920 Summer Olympics           7       Swimming   
5      54255  1992 Summer Olympics          23         Diving   

                            result_location result_detail  start_date  \
0                                  Oostende            na  1920-07-07   
1          Dvorets sporta Izmaylovo, Moskva            na  1980-07-22   
2            Wembley Arena, Wembley, London            na  1948-08-04   
3      Zwemstadion van Antwerpen, Antwerpen            na  1920-08-22   
5  Piscina Municipal de Montjuïc, Barcelona            na  1992-08-02   

     end_date         time  
0  1920-07-09           na  
1          na  14:00-19:00  
2  1948-08-05           na  
3  1920-08-23           na  
5  1992-08-03           na

In [11]:
naOR.shape

(5719, 9)

I find it strange that the result table that is supposed to have the result for which athlete won a 

In [18]:
print(5719/7329)

0.7803247373447947
